scripts/services/map_manager.gd

```diff
@@ -66,12 +78,26 @@ func get_pages_count(page_size):
     return pages_count
 
 func map_exists(map_name):
+    if bundled_maps.has(map_name):
+        return true
     var filepath = self.MAP_PATH + map_name + self.MAP_EXTENSION
     return self.filesystem.file_exists(filepath)
 
 func get_map_data(map_name):
-    var filepath = self.MAP_PATH + map_name + self.MAP_EXTENSION
+    var filepath = get_map_path(map_name)
     return self.filesystem.read_json_from_file(filepath)
 
+func save_map_to_file(filename, content):
+    var filepath = get_map_path(filename)
```

Comment:
If someone happen to name their map the same as a bundled or menu map, it is going to return path to bundled file, where it is not supposed to write, and will most likely fail.

Reply:
Ditto 'self'